benchmarks/time_solve_models.py

```diff
@@ -22,11 +22,328 @@ def prepare_model(model):
+
+
+class TimeSolveSPMORegan2021:
+    params = [True, False]
+    solver = pybamm.CasadiSolver()
+
+    def setup(self, solve_first):
+        self.model = pybamm.lithium_ion.SPM()
+        c_rate = 1
+        tmax = 4000 / c_rate
+        nb_points = 500
+        self.t_eval = np.linspace(0, tmax, nb_points)
+        prepare_model_ORegan2021(self.model)
+        if solve_first:
+            solve_model_once(self.model, TimeSolveSPMORegan2021.solver, self.t_eval)
+
+    def time_solve_model(self, solve_first):
+        TimeSolveSPMORegan2021.solver.solve(self.model, t_eval=self.t_eval)
+
+
+class TimeSolveSPMNCA_Kim2011:
+    params = [True, False]
+    solver = pybamm.CasadiSolver()
+
+    def setup(self, solve_first):
+        self.model = pybamm.lithium_ion.SPM()
+        c_rate = 1
+        tmax = 4000 / c_rate
+        nb_points = 500
+        self.t_eval = np.linspace(0, tmax, nb_points)
+        prepare_model_NCA_Kim2011(self.model)
+        if solve_first:
+            solve_model_once(self.model, TimeSolveSPMNCA_Kim2011.solver, self.t_eval)
+
+    def time_solve_model(self, solve_first):
+        TimeSolveSPMNCA_Kim2011.solver.solve(self.model, t_eval=self.t_eval)
+
+
+class TimeSolveSPMPrada2013:
+    params = [True, False]
+    solver = pybamm.CasadiSolver()
+
+    def setup(self, solve_first):
+        self.model = pybamm.lithium_ion.SPM()
+        c_rate = 1
+        tmax = 4000 / c_rate
+        nb_points = 500
+        self.t_eval = np.linspace(0, tmax, nb_points)
+        prepare_model_Prada2013(self.model)
+        if solve_first:
+            solve_model_once(self.model, TimeSolveSPMPrada2013.solver, self.t_eval)
+
+    def time_solve_model(self, solve_first):
+        TimeSolveSPMPrada2013.solver.solve(self.model, t_eval=self.t_eval)
+
+
+class TimeSolveSPMAi2020:
+    params = [True, False]
+    solver = pybamm.ScikitsDaeSolver()
+
+    def setup(self, solve_first):
+        self.model = pybamm.lithium_ion.SPM()
+        c_rate = 1
+        tmax = 4000 / c_rate
+        nb_points = 500
+        self.t_eval = np.linspace(0, tmax, nb_points)
+        prepare_model_Ai2020(self.model)
+        if solve_first:
+            solve_model_once(self.model, TimeSolveSPMAi2020.solver, self.t_eval)
+
+    def time_solve_model(self, solve_first):
+        TimeSolveSPMAi2020.solver.solve(self.model, t_eval=self.t_eval)
+
+
+class TimeSolveSPMRamadass2004:
+    params = [True, False]
+    solver = pybamm.CasadiSolver()
+
+    def setup(self, solve_first):
+        self.model = pybamm.lithium_ion.SPM()
+        c_rate = 1
+        tmax = 4000 / c_rate
+        nb_points = 500
+        self.t_eval = np.linspace(0, tmax, nb_points)
+        prepare_model_Ramadass2004(self.model)
+        if solve_first:
+            solve_model_once(self.model, TimeSolveSPMRamadass2004.solver, self.t_eval)
+
+    def time_solve_model(self, solve_first):
+        TimeSolveSPMRamadass2004.solver.solve(self.model, t_eval=self.t_eval)
```

Comment:
This applies everywhere (but just spotted it here). You can instead type `self` to refer to the own class, so it is more compact.